modules/dxp/apps/scim/scim-rest-impl/src/main/java/com/liferay/scim/rest/internal/manager/UserManagerImpl.java

```diff
@@ -535,6 +717,85 @@ private ScimUser _getScimUser(long companyId, long userId)
 		return _toScimUser(portalUser);
 	}
 
+	private UserGroup _getUserGroup(long companyId, long groupId)
+		throws AbstractCharonException {
+
+		UserGroup userGroup = _userGroupLocalService.fetchUserGroup(groupId);
+
+		if (userGroup == null) {
+			throw new NotFoundException(
+				"No group found with group ID " + groupId);
+		}
+
+		String groupScimClientId = _getScimClientId(userGroup);
+
+		if (Validator.isNull(groupScimClientId)) {
+			throw new NotFoundException(
+				"No group found with group ID " + groupId);
+		}
+
+		ScimClientOAuth2ApplicationConfiguration
+			scimClientOAuth2ApplicationConfiguration =
+				_getScimClientOAuth2ApplicationConfiguration(companyId);
+
+		String scimClientId = ScimClientUtil.generateScimClientId(
+			scimClientOAuth2ApplicationConfiguration.oAuth2ApplicationName());
+
+		if (!Objects.equals(groupScimClientId, scimClientId)) {
+			throw new ConflictException(
+				"Group was provisioned by another SCIM client");
+		}
+
+		return userGroup;
+	}
+
+	private UserGroup _getUserGroup(
+		long companyId, String externalReferenceCode, long groupId) {
```

Comment:
Here also. `long groupId`

Reply:
And the method should be named `_fetchUserGroup()` because it can return null and doesn't throw exceptions.